src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -267,10 +267,8 @@ Status CoreWorkerMemoryStore::Get(const std::vector<ObjectID> &object_ids,
   }
 
   // Only send block/unblock IPCs for non-actor tasks on the main thread.
-  // TODO(ekl) support non-lifetime resources for direct actor calls.
   bool should_notify_raylet =
-      (raylet_client_ != nullptr && !ctx.CurrentActorIsDirectCall() &&
-       ctx.CurrentThreadIsMain());
+      (raylet_client_ != nullptr && ctx.ShouldReleaseResourcesOnBlockingCalls());
```

Comment:
Is the `raylet_client_` actually `nullptr` under non-error conditions?

Reply:
Yea see the header.